iphone/Maps/Classes/MapDownloader/MWMBaseMapDownloaderViewController.mm

```diff
@@ -501,6 +501,11 @@ - (IBAction)longPress:(UILongPressGestureRecognizer *)sender
 
 - (void)actionSheet:(UIActionSheet *)actionSheet clickedButtonAtIndex:(NSInteger)buttonIndex
 {
+  if (buttonIndex >= actionSheet.numberOfButtons)
```

Comment:
Тогда нужно было написать в фиксе комментарий про 7 ось и про то, что такого не должно быть.
А для какого диалога приходил такой индекс? Локально воспроизводить пробовали?

> On Apr 8, 2016, at 15:51, Vlad Mihaylenko <notifications@github.com> wrote:
> 
> In iphone/Maps/Classes/MapDownloader/MWMBaseMapDownloaderViewController.mm <https://github.com/mapsme/omim/pull/2779#discussion_r59019623>:
> 
> > @@ -501,6 +501,11 @@ - (IBAction)longPress:(UILongPressGestureRecognizer *)sender
> >  
> >  - (void)actionSheet:(UIActionSheet *)actionSheet clickedButtonAtIndex:(NSInteger)buttonIndex
> >  {
> > +  if (buttonIndex >= actionSheet.numberOfButtons)
> этот метод делегата зовет система. на 7ой оси, из крэш лога, нам приходил огромный индекс и пустой массив кнопок. в чем баг непонятно. он воспроизводился только на 7ой оси, а там сейчас многое работает некорректно. как еще это пофиксить мы не придумали.
> из того, как работает эта апишка, такое поведение невозможно.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2779/files/362adb4fcf0abf0f56f27377d9292e1ce953f00f#r59019623>